src/worker/tasks/sendTransactionWorker.ts

```diff
@@ -81,11 +81,48 @@ const handler: Processor<any, void, string> = async (job: Job<string>) => {
   // null = the transaction attemped to resend but was not needed. Ignore.
   // A thrown exception indicates a retry-able error occurred (e.g. RPC outage).
   let resultTransaction: SentTransaction | ErroredTransaction | null;
-  if (transaction.status === "queued") {
+
+  let populatedTransaction:
+    | Awaited<ReturnType<typeof toSerializableTransaction>>
+    | undefined;
+
+  const chain = await getChain(transaction.chainId);
+  // Populate the transaction to resolve gas values.
+  // This call throws if the execution would be reverted.
+  // The nonce is _not_ set yet.
+  try {
+    populatedTransaction = await toSerializableTransaction({
+      from: getChecksumAddress(transaction.from),
+      transaction: {
+        client: thirdwebClient,
+        chain,
+        ...transaction,
+        // Stub the nonce because it will be overridden later.
+        nonce: 1,
+      },
+    });
+  } catch (e: unknown) {
+    // If the transaction will revert, error.message contains the human-readable error.
+    const errorMessage = (e as Error)?.message ?? `${e}`;
+    job.log(`Failed to estimate gas: ${errorMessage}`);
```

Comment:
Nit: would change this msg even if it's internal to something like "tx validation failed" or something